Code/Tools/AssetProcessor/Platform/Linux/native/FileWatcher/FileWatcher_linux.cpp

```diff
@@ -20,150 +23,151 @@ static constexpr size_t s_iNotifyMaxEntries = 1024 * 16;         // Control the
 static constexpr size_t s_iNotifyEventSize = sizeof(struct inotify_event);
 static constexpr size_t s_iNotifyReadBufferSize = s_iNotifyMaxEntries * s_iNotifyEventSize;
 
-struct FolderRootWatch::PlatformImplementation
+bool FileWatcher::PlatformImplementation::Initialize()
 {
-    PlatformImplementation() = default;
-
-    int                         m_iNotifyHandle = -1;
-    QMutex                      m_handleToFolderMapLock;
-    QHash<int, QString>         m_handleToFolderMap;
-
-    bool Initialize()
+    if (m_iNotifyHandle < 0)
     {
+        // The CLOEXEC flag prevents the inotify watchers from copying on fork/exec
+        m_iNotifyHandle = inotify_init1(IN_CLOEXEC);
+
         if (m_iNotifyHandle < 0)
         {
             // The CLOEXEC flag prevents the inotify watchers from copying on fork/exec
             m_iNotifyHandle = inotify_init1(IN_CLOEXEC);
+            const auto err = errno;
```

Comment:
you probably will need a [[maybe_unused]] here because AZ_Warning is not processed in release builds.